connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java

```diff
@@ -0,0 +1,242 @@
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(Enclosed.class)
+public class RestClientTest {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<TestDTO> TEST_TYPE = new TypeReference<TestDTO>() {
+    };
+    private static final SecretKey MOCK_SECRET_KEY = getMockSecretKey();
+
+    private static void assertIsInternalServerError(ConnectRestException e) {
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), e.statusCode());
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), e.errorCode());
+    }
+
+    private static SecretKey getMockSecretKey() {
+        SecretKey mockKey = mock(SecretKey.class);
+        when(mockKey.getFormat()).thenReturn("RAW");// supported format by
+        when(mockKey.getEncoded()).thenReturn("SomeKey".getBytes(StandardCharsets.UTF_8));
+        return mockKey;
+    }
+
+    private static RestClient.HttpResponse<TestDTO> httpRequest(HttpClient httpClient, String requestSignatureAlgorithm) {
+        return RestClient.httpRequest(
+                httpClient,
+                "https://localhost:1234/api/endpoint",
+                "GET",
+                null,
+                new TestDTO("requestBodyData"),
+                TEST_TYPE,
+                MOCK_SECRET_KEY,
+                requestSignatureAlgorithm);
+    }
+
+    private static RestClient.HttpResponse<TestDTO> httpRequest(HttpClient httpClient) {
+        String validRequestSignatureAlgorithm = "HmacSHA1";
+        return httpRequest(httpClient, validRequestSignatureAlgorithm);
+    }
+
+
+    @RunWith(Parameterized.class)
+    public static class RequestFailureParameterizedTest {
+        private final HttpClient httpClient = mock(HttpClient.class);
+
+        @Parameterized.Parameter
+        public Throwable requestException;
+
+        @Parameterized.Parameters
+        public static Collection<Object[]> requestExceptions() {
+            return Arrays.asList(new Object[][]{
+                    {new InterruptedException()},
+                    {new ExecutionException(null)},
+                    {new TimeoutException()}
+            });
+        }
+
+        private static Request buildThrowingMockRequest(Throwable t) throws ExecutionException, InterruptedException, TimeoutException {
+            Request req = mock(Request.class);
+            when(req.send()).thenThrow(t);
+            return req;
+        }
+
+        @Test
+        public void testFailureDuringRequestCausesInternalServerError() throws Exception {
+            Request request = buildThrowingMockRequest(requestException);
+            when(httpClient.newRequest(anyString())).thenReturn(request);
+            ConnectRestException e = assertThrows(ConnectRestException.class, () -> httpRequest(httpClient));
+            assertIsInternalServerError(e);
```

Comment:
We should also make sure that the cause of the exception is properly tracked:
```suggestion
            assertIsInternalServerError(e);
            assertEquals(requestException, e.getCause());
```

Reply:
fixed